compiler-core/src/analyse.rs

```diff
@@ -1102,6 +1110,35 @@ impl<'a, A> ModuleAnalyzer<'a, A> {
             hydrator.disallow_new_type_variables();
             let type_ = hydrator.type_from_ast(resolved_type, environment, &mut self.problems)?;
 
+            if let Type::Named {
+                module,
+                name: type_name,
+                args,
+                ..
+            } = type_.as_ref()
+            {
+                // If the alias or type itself have any parameters, it requires a
+                // lot of work to reconstruct and correctly print the alias,
+                // so we just don't register it.
+                //
+                // Maybe in future we could do this extra work, but for the time
+                // being, it's simpler not to.
+                if parameters.is_empty() && args.is_empty() {
+                    environment.names.named_type_in_scope(
+                        module.clone(),
+                        type_name.clone(),
+                        name.clone(),
+                    );
+                } else {
+                    // We need to make sure we register the type as existing, even if we don't define
+                    // the underlying type for printing, because these types do still affect printing,
+                    // for shadowing prelude types, even if they aren't printed at all.
```

Comment:
Sorry, what does this mean? How does it affect printing?

Reply:
This has to do with the way prelude types are handled, which are discussed above. I'll wait until further feedback on that to elaborate, because this may become redundant.